thorlabs_apt_protocol/functions.py

```diff
@@ -2183,3 +2183,198 @@ def pol_set_params(
 
 def pol_req_params(dest: int, source: int):
     return _pack(0x0531, dest, source)
+
+
+def mot_set_mottrigioconfig(
+    dest: int,
+    source: int,
+    chan_ident: int,
+    trig_in_mode: int,
+    trig_in_polarity: int,
+    trig_in_source: int,
+    trig_out_mode: int,
+    trig_out_polarity: int,
+    start_pos_fwd: int,
+    interval_fwd: int,
+    num_pulses_fwd: int,
+    start_pos_rev: int,
+    interval_rev: int,
+    num_pulses_rev: int,
+    pulse_width: int,
+    num_cycles: int,
+) -> bytes:
+    data = struct.pack(
+        "<6H8lH",
+        chan_ident,
+        trig_in_mode,
+        trig_in_polarity,
+        trig_in_source,
+        trig_out_mode,
+        trig_out_polarity,
+        start_pos_fwd,
+        interval_fwd,
+        num_pulses_fwd,
+        start_pos_rev,
+        interval_rev,
+        num_pulses_rev,
+        pulse_width,
+        num_cycles,
+        0,
+    )
+    return _pack(0x0260, dest, source, data=data)
+
+
+def mot_req_mottrigioconfig(dest: int, source: int, chan_ident: int) -> bytes:
+    return _pack(0x0261, dest, source, param1=chan_ident)
+
+
+def mot_set_ioconfig(
+    dest: int, source: int, io_port: int, mode: int, out_source: int
+) -> bytes:
+    data = struct.pack("<HHH", io_port, mode, out_source)
+    return _pack(0x0263, dest, source, data=data)
+
+
+def mot_req_ioconfig(dest: int, source: int, chan_ident: int, io_port: int) -> bytes:
+    return _pack(0x0264, dest, source, param1=chan_ident, param2=io_port)
+
+
+def mot_set_auxioconfig(
+    dest: int, source: int, out_port: int, mode: int, sw_source: int
+) -> bytes:
+    data = struct.pack("<HHH", out_port, mode, sw_source)
+    return _pack(0x0266, dest, source, data=data)
+
+
+def mot_req_auxioconfig(
+    dest: int, source: int, chan_ident: int, out_port: int
+) -> bytes:
+    return _pack(0x0267, dest, source, param1=chan_ident, param2=out_port)
+
+
+def mot_set_analogmonitorconfig(
+    dest: int,
+    source: int,
+    monitor: int,
+    motor_channel: int,
+    sys_var: int,
+    scale: int,
+    offset: int,
+) -> bytes:
+    data = struct.pack("<HHHll", monitor, motor_channel, sys_var, scale, offset)
+    return _pack(0x0269, dest, source, data=data)
+
+
+def mot_req_analogmonitorconfig(
+    dest: int, source: int, chan_ident: int, io_port: int
+) -> bytes:
+    return _pack(0x0270, dest, source, param1=chan_ident, param2=io_port)
```

Comment:
```suggestion
def mot_req_analogmonitorconfig(
    dest: int, source: int, chan_ident: int) -> bytes:
    return _pack(0x0270, dest, source, param1=chan_ident)
```

I do not have this hardware, so feel free to reassert that the original is correct, I just don't see anything about that parameter in the published protocol (wouldn't be the only inaccuracy, though)

Ref pg. 167 of the Version 33 pdf (current)

Reply:
I found in testing that the second parameter field was the one that selected which output monitor was being addressed. 0x01 returned data on monitor 1, while 0x02 returned data on monitor 2, regardless of the value in chan_ident.  0x00 returned nothing, so their example message wouldn't work (also this message is sent to the controller, address 0x11, not the individual channel controllers).